### PR TITLE
Fixing appeal details

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects #6540

### Description
In the last refactor we created an `appeals` and `appealDetails` section of the state. However the `appeals` section which contains only basic appeal information was structured without the `attributes` sub-object and the sub keys were in camel case. The `appealDetails` did have the `attributes` sub-object and the sub keys were still snake cased. Finally, there was overlap between the `appealDetails` and `appeals` section. This removes that overlap and creates consistency in how we represent our state.

We've created two new selectors: `appealsWithDetailsSelector` and `appealWithDetailSelector` to get the combined basic and detailed appeal information.

### Testing Plan
1. All the tests still pass and a manual walkthrough shows no changes.

### CODE REVIEW

File: client/app/queue/utils.js [hunk 1]
```diff
@@ -80,17 +75,73 @@ export const associateTasksWithAppeals =
...
-export const prepareAppealDetailsForStore =
-  (appeals: Array<LegacyAppeal>):
-    LegacyAppeals => {
-    return _.pickBy(_.keyBy(appeals, (appeal) => appeal.attributes.external_id), (appeal) => appeal);
+export const prepareAppealForStore =
+  (appeals: Array<Object>):
+    { appeals: BasicAppeals, appealDetails: AppealDetails } => {
+
+    const appealHash = appeals.reduce((accumulator, appeal) => {
+      if (!accumulator[appeal.attributes.external_id]) {
```
[the remark this answers is not in the document]
Comment (mdbenjam) [reply client/app/queue/utils.js]: Ah good call, I think we can remove this. I copied this from some other code, unclear it's definitely needed there either...

File: client/app/queue/CaseDetailsLink.jsx [hunk 2]
```diff
@@ -15,11 +15,11 @@ export default class CaseDetailsLink extends React.PureComponent {
     } = this.props;
 
     // For now while only the basic appeal info is named properly this is necessary. To be removed later.
-    const isPaperCase = appeal.isPaperCase || (appeal.attributes && appeal.attributes.paper_case);
+    const isPaperCase = appeal.isPaperCase;
```
Comment (lowellrex) [inline client/app/queue/CaseDetailsLink.jsx]: Can we remove the comment above this line now?

Comment (lowellrex) [reply client/app/queue/CaseDetailsLink.jsx]: Also, I think this variable might be unnecessary now. We could just make line 27: `{appeal.isPaperCase && ...`

File: client/app/queue/AddEditIssueView.jsx [hunk 3]
```diff
@@ -110,7 +110,7 @@ class AddEditIssueView extends React.Component {
       }
     };
     const issueIndex = _.map(issues, 'vacols_sequence_id').indexOf(issue.vacols_sequence_id);
-    const url = `/appeals/${appeal.id}/issues`;
+    const url = `/appeals/${appeal.externalId}/issues`;
```
Comment (lowellrex) [inline client/app/queue/AddEditIssueView.jsx]: Is `BasicAppeal.id` the same thing as `AppealDetail.externalId`?

Comment (lowellrex) [reply client/app/queue/AddEditIssueView.jsx]: No, but this change in behaviour is supported by the associated change here: https://github.com/department-of-veterans-affairs/caseflow/pull/6544/files#r209049344

**Pull request merged**